src/controller/c/speaker.c

```diff
@@ -298,6 +312,43 @@ static void speaker_toggle_remote(WbDevice *d, WbRequest *r) {
   // nothing to do.
 }
 
+/* Try to read sound file contents if it is found by given path. */
+static void speaker_try_load_sound_local(Sound *sound, const char *sound_file_name) {
+  FILE *fp;
+
+  sound->upload_data = NULL;
+  sound->upload_size = 0;
+
+  if ((fp = fopen(sound_file_name, "rb")) != NULL) {
+    if (fseek(fp, 0L, SEEK_END) == 0) {
+      const long file_size = ftell(fp);
+      if (file_size != -1) {
+        if (fseek(fp, 0L, SEEK_SET) == 0) {
+          size_t consumed = 0U;
+          sound->upload_data = malloc(file_size);
+          sound->upload_size = (int)file_size;
+          while (consumed < file_size) {
+            const size_t read = fread(&sound->upload_data[consumed], sizeof(unsigned char), file_size - consumed, fp);
+            if (ferror(fp)) {
+              free(sound->upload_data);
+              sound->upload_data = NULL;
+              sound->upload_size = 0;
+              break;
+            }
+            consumed += read;
+          }
+        }
+      }
+    }
+
+    if (NULL == sound->upload_data) {
+      fprintf(stderr, "Warning: %s() failed to read sound file '%s' :", __FUNCTION__, sound_file_name);
+      perror(NULL);
+    }
```

Comment:
I am not sure we want to display a warning here, as now, it could be perfect use case to play a sound that is only on the server side, i.e., for example relative to the robot PROTO folder. A warning should be displayed if the sound if not found either on the controller side nor on the Webots side.

Reply:
This test is inside the if block of a succesfull fopen() on the file. It means the file was found by the controller, but something odd happend during reading it.  It is unlikely to appear but usefull for debug why a sound didn't play, also is was on the controller end.

If the file isn't found on either end, Webots shows the error with the controller provided path. This got fixed in PR #6593.

Reply:
Sorry, I didn't notice. My Bad.